deployment/common/changeset/link_transfer_timelock_test.go

```diff
@@ -0,0 +1,121 @@
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+	"github.com/smartcontractkit/chainlink/deployment/common/changeset"
+	"github.com/smartcontractkit/chainlink/deployment/common/types"
+	"github.com/smartcontractkit/chainlink/deployment/environment/memory"
+)
+
+// TestLinkTransferTimelock tests the LinkTransferTimelock changeset by
+func TestLinkTransferTimelock(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	lggr := logger.TestLogger(t)
+	cfg := memory.MemoryEnvironmentConfig{
+		Nodes:  1,
+		Chains: 2,
+	}
+	env := memory.NewMemoryEnvironment(t, lggr, zapcore.DebugLevel, cfg)
+	chainSelector := env.AllChainSelectors()[0]
+	chain := env.Chains[chainSelector]
+	config := changeset.SingleGroupMCMS(t)
+
+	// Deploy MCMS and Timelock
+	env, err := changeset.ApplyChangesets(t, env, nil, []changeset.ChangesetApplication{
+		{
+			Changeset: changeset.WrapChangeSet(changeset.DeployLinkToken),
+			Config:    []uint64{chainSelector},
+		},
+		{
+			Changeset: changeset.WrapChangeSet(changeset.DeployMCMSWithTimelock),
+			Config: map[uint64]types.MCMSWithTimelockConfig{
+				chainSelector: {
+					Canceller:        config,
+					Bypasser:         config,
+					Proposer:         config,
+					TimelockMinDelay: big.NewInt(0),
+				},
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	addrs, err := env.ExistingAddresses.AddressesForChain(chainSelector)
+	require.NoError(t, err)
+	require.Len(t, addrs, 6)
+
+	mcmsState, err := changeset.MaybeLoadMCMSWithTimelockState(chain, addrs)
+	require.NoError(t, err)
+	linkState, err := changeset.MaybeLoadLinkTokenState(chain, addrs)
+	require.NoError(t, err)
+	timelockAddress := mcmsState.Timelock.Address()
+
+	// Mint some funds
+	// grant minter permissions
+	tx, err := linkState.LinkToken.GrantMintRole(chain.DeployerKey, chain.DeployerKey.From)
+	require.NoError(t, err)
+	_, err = deployment.ConfirmIfNoError(chain, tx, err)
+	require.NoError(t, err)
+
+	tx, err = linkState.LinkToken.Mint(chain.DeployerKey, timelockAddress, big.NewInt(750))
+	require.NoError(t, err)
+	_, err = deployment.ConfirmIfNoError(chain, tx, err)
+	require.NoError(t, err)
+
+	timelocks := map[uint64]*changeset.TimelockExecutionContracts{
+		chainSelector: {
+			Timelock:  mcmsState.Timelock,
+			CallProxy: mcmsState.CallProxy,
+		},
+	}
+
+	// Apply the changeset
+	_, err = changeset.ApplyChangesets(t, env, timelocks, []changeset.ChangesetApplication{
+		// the changeset produces proposals, ApplyChangesets will sign & execute them.
+		// in practice, signing and executing are separated processes.
+		{
+			Changeset: changeset.WrapChangeSet(changeset.LinkTransferTimelock),
+			Config: &changeset.LinkTransferTimelockConfig{
+				Transfers: map[uint64][]changeset.LinkTransfer{
+					chainSelector: {
+						{
+							To:    chain.DeployerKey.From,
+							Value: big.NewInt(500),
+						},
+					},
+				},
+				UseMCMS: true,
```

Comment:
need a test case for when false

Reply:
added here https://github.com/smartcontractkit/chainlink/pull/15512/commits/f889dfc35991573e31c5a56015b3780ff673a9a5